_includes/exhibits.js

```diff
@@ -0,0 +1,1929 @@
+    }
+
+    // Redraw design
+    if(redraw) {
+      // Update OpenSeadragon
+      this.activateViewport();
+      newMarkers(this.tileSources, this.group);
+      // Redraw HTML Menus
+      this.addChannelLegends();
+      this.addGroups();
+      this.newStories();
+
+      // back and forward
+      $('.step-back').click(this, function(e) {
+        const THIS = e.data;
+        THIS.w -= 1;
+        THIS.pushState();
+      });
+      $('.step-next').click(this, function(e) {
+        const THIS = e.data;
+        THIS.w += 1;
+        THIS.pushState();
+      });
+
+      // Waypoint-specific Copy Buttons
+      const STATE = this;
+      $('.edit_copy_button').each(function() {
+        newCopyYamlButton.call(this, STATE);
+      });
+    }
+
+    // Update text
+    const v = this.v;
+    const o = this.o;
+    $('.edit_zoom').text(show5(v[0]));
+    $('.edit_panx').text(show5(v[1]));
+    $('.edit_pany').text(show5(v[2]));
+    $('.edit_x').text(show5(o[0]));
+    $('.edit_y').text(show5(o[1]));
+    $('.edit_w').text(show5(o[2]));
+    $('.edit_h').text(show5(o[3]));
+
+    // Based on control keys
+    const editing = this.editing;
+    const drawing = this.drawing;
+
+    // Based on search keys
+    displayOrNot('#draw-switch a', !editing);
+    displayOrNot('.show-if-edit', this.edit);
+    activeOrNot('#view-switch', !editing);
+    activeOrNot('#edit-switch', editing);
+    displayOrNot('#story-nav', !editing);
+    displayOrNot('.edit-item', editing);
+
+    classOrNot('#story-nav', !editing, 'round-nav');
+    classOrNot('#edit-menu', editing, 'round-nav');
+
+    toggleCursor('crosshair', drawing);
+
+    greenOrWhite('.draw-switch *', drawing);
+    //greenOrWhite('#edit-switch *', editing);
+  },
+
+  makeUrl: function(hashKeys, searchKeys) {
+    const root = this.location('pathname');
+    const hash = this.makeHash(hashKeys);
+    const search = this.makeSearch(searchKeys);
+    return  root + search + hash;
+  },
+
+  makeHash: function(hashKeys, state) {
+    if (state  == undefined) {
+      state = this;
+    }
+    if (hashKeys == undefined) {
+      hashKeys = this.hashKeys;
+    }
+    const hash = serialize(hashKeys, state, '#');
+    return hash? '#' + hash : '';
+  },
+
+  makeSearch: function(searchKeys, state) {
+    if (state  == undefined) {
+      state = this;
+    }
+    if (searchKeys == undefined) {
+      searchKeys = this.searchKeys;
+    }
+    const search = serialize(searchKeys, state, '&');
+    return search? '?' + search : '';
+  },
+
+  /*
+   * User intercation
+   */
+  normalize: function(pixels) {
+    const vp = this.viewer.viewport;
+    const norm = vp.viewerElementToViewportCoordinates;
+    return norm.call(vp, pixels);
+  },
+  drawLowerBounds: function(position) {
+    const wh = [0, 0];
+    const new_xy = [
+      position.x, position.y
+    ];
+    this.o = new_xy.concat(wh);
+    this.newView(false);
+  },
+  computeBounds: function(value, start, len) {
```

Comment:
This function determines how to modify the overlay box based on the user's mouse position. It may increase the size of the box or change the origin of the box.